tests/lib/test_openstack_flavor.py

```diff
@@ -0,0 +1,140 @@
+import unittest
+from unittest.mock import NonCallableMock, Mock, MagicMock, call
+from openstack_api.openstack_flavor import OpenstackFlavor
+
+
+# pylint:disable=too-many-public-methods
+class OpenstackFlavorTests(unittest.TestCase):
+    """
+    Runs various tests to ensure we are using the Openstack
+    Flavor module in the expected way
+    """
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.mocked_connection = MagicMock()
+        self.instance = OpenstackFlavor(self.mocked_connection)
+        self.api = self.mocked_connection.return_value.__enter__.return_value
+
+    def test_list_flavor(self):
+        """
+        Test that we can retrieve a list of flavors from a specific cloud
+        """
+        mocked_name = NonCallableMock()
+        flavors = self.instance.list_flavor(mocked_name)
+        assert flavors == self.api.list_flavors.return_value
+        self.mocked_connection.assert_called_once_with(mocked_name)
+
+    def test_get_flavor(self):
+        """
+        Test that we can get information for a specific flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor_name = NonCallableMock()
+        flavor = self.instance.get_flavor(mocked_cloud_name, mocked_flavor_name)
+
+        assert flavor == self.api.get_flavor.return_value
+
+        self.api.get_flavor.assert_called_once_with(mocked_flavor_name)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
+
+    def test_get_missing_flavors_calls_list_flavor(self):
```

Comment:
`test_get_missing_flavors_calls_list_flavor_with_correct_clouds`
And the docstring is incorrect, as this test is checking we pass the source and dest cloud in the correct order rather than checking the diff in the next test